src/sdf/src/sdf.cpp

```diff
@@ -356,5 +361,156 @@ Mesh LevelSet(std::function<float(glm::vec3)> sdf, Box bounds, float edgeLength,
   out.triVerts.insert(out.triVerts.end(), triVerts.begin(), triVerts.end());
   return out;
 }
+
+/**
+ * Similar to LevelSet, this constructs a level-set Mesh from the input
+ * Signed-Distance Function (SDF). This variant feeds an std:vector of points
+ * to the function, allowing the function to compute signed distances using
+ * its own threading/parallelization paradigm.
+ *
+ * @param sdf The signed-distance functor, containing this function signature:
+ * `float operator()(glm::vec3 point)`, which returns the
+ * signed distance of a given point in R^3. Positive values are inside,
+ * negative outside.
+ * @param bounds An axis-aligned box that defines the extent of the grid.
+ * @param edgeLength Approximate maximum edge length of the triangles in the
+ * final result. This affects grid spacing, and hence has a strong effect on
+ * performance.
+ * @param level You can inset your Mesh by using a positive value, or outset
+ * it with a negative value.
+ * @param canParallel Parallel policies violate will crash language runtimes
+ * with runtime locks that expect to not be called back by unregistered threads.
+ * This allows bindings use LevelSet despite being compiled with MANIFOLD_PAR
+ * active.
+ * @return Mesh This class does not depend on Manifold, so it just returns a
+ * Mesh, but it is guaranteed to be manifold and so can always be used as
+ * input to the Manifold constructor for further operations.
+ */
+Mesh LevelSetBatch(
+    std::function<std::vector<float>(std::vector<glm::vec3>)> sdf, Box bounds,
+    float edgeLength, float level, bool canParallel) {
+  Mesh out;
+
+  const glm::vec3 dim = bounds.Size();
+  const float maxDim = std::max(dim[0], std::max(dim[1], dim[2]));
+  const glm::ivec3 gridSize(dim / edgeLength);
+  const glm::vec3 spacing = dim / (glm::vec3(gridSize));
+
+  const Uint64 maxMorton = MortonCode(glm::ivec4(gridSize + 1, 1));
+
+  // Parallel policies violate will crash language runtimes with runtime locks
+  // that expect to not be called back by unregistered threads. This allows
+  // bindings use LevelSet despite being compiled with MANIFOLD_PAR
+  // active.
+  const auto pol = canParallel ? autoPolicy(maxMorton) : ExecutionPolicy::Seq;
+
+  int tableSize = glm::min(
+      2 * maxMorton, static_cast<Uint64>(10 * glm::pow(maxMorton, 0.667)));
+  HashTable<GridVert, identity> gridVerts(tableSize);
+  Vec<glm::vec3> vertPos(gridVerts.Size() * 7, glm::vec3(0.0));
+
+  // BEGIN DIVERGENCE ----------------------------------------------------------
```

Comment:
Can we reduce duplication with `LevelSet`?

Reply:
I'm not sure what the best way to do this is...
- The function signature is different (Though, I could make _all_ LevelSet calls ask for batched queries, and have a skeleton function overload that replicates the existing signature?  More perf/robustness-testing needed there.)
- It's tough to combine the logic because the changes touch almost every line, even if it's superficial...
- Open to thoughts here.